[libuuid]Add warning to non-unix platform when building.

JackBoosY opened the pull request:
After reviewing the issue #3852, I found that there are many Linux functions in the source code. Obviously libuuid does not support non-Linux systems, so I added a warning that it will fail directly when buiding under non-Linux systems.

Commit:
[jemalloc]Fix dll generated when generating static library

Commit:
Revert "[jemalloc]Fix dll generated when generating static library"

This reverts commit a833bf5593be181e065af097732c78cb810888fd.

Commit:
Merge branch 'master' of https://github.com/Microsoft/vcpkg

Commit:
[libuuid]Add warning to non-unix platform when building.

myd7349 commented:
How about macOS?

Rastaban commented on ports/libuuid/portfile.cmake:
```diff
@@ -1,5 +1,9 @@
 include(vcpkg_common_functions)
 
+if (NOT VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
`NOT (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin")`

vicroms replied:
MacOS has its own `uuid.h` file, this library actually causes conflicts in `vtk`. So we shouldn't allow it to be installed for Darwin either.

See https://github.com/Microsoft/vcpkg/pull/5722#issuecomment-477687011

JackBoosY replied:
VTK and ATK always have problem. I think we should solve these problems quickly.

Commit:
[libuuid]modified the macro that judges the build platform.

JackBoosY commented:
@myd7349 I'm not sure, sorry.

vicroms commented on ports/libuuid/portfile.cmake:
```diff
@@ -1,5 +1,9 @@
 include(vcpkg_common_functions)
 
+if (NOT (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin"))
```
Revert to only allow Linux, MacOS provides its own implementation of `uuid.h`. 
See https://github.com/Microsoft/vcpkg/pull/5722#issuecomment-477687011

vicroms replied:
Note that `azure-storage-cpp` has a dependency on `libuuid(!windows&!uwp)` that should be changed to `libuuid(!windows&!uwp&!osx)`

Rastaban replied:
You are right, although this port builds on Mac it should not be used.  This change will help the 'vtk' port be more stable in the CI tests.

Rastaban replied:
In the interest of getting this in soon so it can clear up other PRs with flakey vtk failures I made the suggested updates.

Rastaban committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg into dev/philc/5844

Rastaban committed:
[azure-storage-cpp/libuuid] disable libuuid on OSX in favor of the built in version

Rastaban commented:
The change to azure-storage-cpp causes a build failure on Mac, working on a patch for the port to use the header that is part of the OS SDK.

Rastaban committed:
[azure-storage-cpp] on osx use sdk uuid.h

Rastaban committed:
[azure-storage-cpp] add gettext to Build-Depends

Rastaban committed:
[azure-storage-cpp] Remove old patch

Rastaban committed:
[libuuid] allow osx

We recommend using the built in uuid.h on osx, but for now won't disallow the use of the libuuid port.
